### DISCUSSION TITLE
Newbie Sample Code for the Custom page

### DISCUSSION THREAD

**Discussion by MarJose123** [filamentphp/filament#3530] [Help]
Can you help me or guide me on how I can create a fields on the custom page, do I need to create the form under the class on my filament page ?

I'm creating a custom page without the resource.

**Comment by MarJose123** [comment]
I tried this code but it didn't reflect on my page.

```
    public static function form(Form $form): Form
    {
        return $form
            ->schema([

                TextInput::make('first_name')->required(),
            ]);
    }
```

**Reply by mikrosmile** [reply]
In your custom page you should have generated view, for example:
```
protected static string $view = 'filament.pages.company';
```

in Page class define your form elements 
```
    protected function getFormSchema(): array
    {
        return [
                TextInput::make('name')
        ];
    }
```

in view blade
```
<x-filament::page>
{{$this->form}}
</x-filament::page>
```

The function will return all your elements from schema. But you need to define actions for buttons and etc if you need to store some data to model. For now, it will just display what you've put there and elements itself.

**Reply by MarJose123** [reply]
This will be a good for start up. thanks.

**Comment by danharrin** [comment]
Follow the Getting Started page of the Form Builder docs, its slightly different to resources. The page class is the Livewire component.

**Reply by MarJose123** [reply]
> Follow the Getting Started page of the Form Builder docs, its slightly different to resources. The page class is the Livewire component.

Is there a chance we can update the documentation and make it more friendly with newbie? like sample start  up code snippet.
